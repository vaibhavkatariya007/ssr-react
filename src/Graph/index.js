import React from 'react';
import {Line, defaults} from 'react-chartjs-2';
import './Graph.css';

defaults.global.elements.line.borderWidth = 1;
defaults.global.elements.line.fill = false;
defaults.scale.gridLines.drawTicks = false;
defaults.scale.ticks.padding = 15;

const Graph = ({data}) => (
  <div className="timeline-chart">
    <Line
      data={data}
      height={350}
      options={{
        legend: {
          position: 'bottom',
          labels: {
            boxWidth: 10,
          },
        },
        layout: {
          padding: {left: 0, right: 0, top: 0, bottom: 0},
        },
        maintainAspectRatio: false,
      }}
    />
  </div>
);

export default Graph;
