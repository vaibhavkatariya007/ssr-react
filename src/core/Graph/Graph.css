/*** -----------------------------------------

              TIMELINE CHART

----------------------------------------- ***/

.timeline-chart {
  margin-top: 35px;
}

.timeline-chart .chartjs-render-monitor {
  height: 350px !important;
}
