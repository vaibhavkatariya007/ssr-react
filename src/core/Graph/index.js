import React from 'react';
import { Line, defaults } from 'react-chartjs-2';
import './Graph.css';

defaults.global.elements.line.borderWidth = 1;
defaults.global.elements.line.fill = false;
defaults.scale.gridLines.drawTicks = false;
defaults.scale.ticks.padding = 15;

const Graph = ({ data }) => (
  <div className="timeline-chart">
    <Line
      data={data}
      height={350}
      options={{
        legend: {
          display: false,
          position: 'left',
          labels: {
            boxWidth: 0,
          },
        },
        layout: {
          padding: { left: 0, right: 0, top: 0, bottom: 0 },
        },
        maintainAspectRatio: false,
        scales: {
          yAxes: [
            {
              barPercentage: 0.2,
              maintainAspectRatio: false,
              scaleLabel: {
                display: true,
                labelString: 'Votes',
              },
            },
          ],
          xAxes: [
            {
              gridLines: {
                display: false,
              },
              scaleLabel: {
                display: true,
                labelString: 'ID',
              },
            },
          ],
        },
      }}
    />
  </div>
);

export default Graph;
