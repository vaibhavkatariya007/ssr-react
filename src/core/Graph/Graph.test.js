import React from 'react';
import { shallow, mount } from 'enzyme';
import Graph from './';
describe('Testing Graph Component', () => {
  const component = shallow(<Graph data={{}} />);

  test('Graph Component exists', () => {
    expect(component.exists()).toBe(true);
  });
});
