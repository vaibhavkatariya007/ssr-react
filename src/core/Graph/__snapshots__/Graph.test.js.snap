// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Graph Component Graph should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Graph />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Line
        height={350}
        options={
          Object {
            "layout": Object {
              "padding": Object {
                "bottom": 0,
                "left": 0,
                "right": 0,
                "top": 0,
              },
            },
            "legend": Object {
              "display": false,
              "labels": Object {
                "boxWidth": 0,
              },
              "position": "left",
            },
            "maintainAspectRatio": false,
            "scales": Object {
              "xAxes": Array [
                Object {
                  "gridLines": Object {
                    "display": false,
                  },
                  "scaleLabel": Object {
                    "display": true,
                    "labelString": "ID",
                  },
                },
              ],
              "yAxes": Array [
                Object {
                  "barPercentage": 0.2,
                  "maintainAspectRatio": false,
                  "scaleLabel": Object {
                    "display": true,
                    "labelString": "Votes",
                  },
                },
              ],
            },
          }
        }
      />,
      "className": "timeline-chart",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "data": undefined,
        "height": 350,
        "options": Object {
          "layout": Object {
            "padding": Object {
              "bottom": 0,
              "left": 0,
              "right": 0,
              "top": 0,
            },
          },
          "legend": Object {
            "display": false,
            "labels": Object {
              "boxWidth": 0,
            },
            "position": "left",
          },
          "maintainAspectRatio": false,
          "scales": Object {
            "xAxes": Array [
              Object {
                "gridLines": Object {
                  "display": false,
                },
                "scaleLabel": Object {
                  "display": true,
                  "labelString": "ID",
                },
              },
            ],
            "yAxes": Array [
              Object {
                "barPercentage": 0.2,
                "maintainAspectRatio": false,
                "scaleLabel": Object {
                  "display": true,
                  "labelString": "Votes",
                },
              },
            ],
          },
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Line
          height={350}
          options={
            Object {
              "layout": Object {
                "padding": Object {
                  "bottom": 0,
                  "left": 0,
                  "right": 0,
                  "top": 0,
                },
              },
              "legend": Object {
                "display": false,
                "labels": Object {
                  "boxWidth": 0,
                },
                "position": "left",
              },
              "maintainAspectRatio": false,
              "scales": Object {
                "xAxes": Array [
                  Object {
                    "gridLines": Object {
                      "display": false,
                    },
                    "scaleLabel": Object {
                      "display": true,
                      "labelString": "ID",
                    },
                  },
                ],
                "yAxes": Array [
                  Object {
                    "barPercentage": 0.2,
                    "maintainAspectRatio": false,
                    "scaleLabel": Object {
                      "display": true,
                      "labelString": "Votes",
                    },
                  },
                ],
              },
            }
          }
        />,
        "className": "timeline-chart",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "data": undefined,
          "height": 350,
          "options": Object {
            "layout": Object {
              "padding": Object {
                "bottom": 0,
                "left": 0,
                "right": 0,
                "top": 0,
              },
            },
            "legend": Object {
              "display": false,
              "labels": Object {
                "boxWidth": 0,
              },
              "position": "left",
            },
            "maintainAspectRatio": false,
            "scales": Object {
              "xAxes": Array [
                Object {
                  "gridLines": Object {
                    "display": false,
                  },
                  "scaleLabel": Object {
                    "display": true,
                    "labelString": "ID",
                  },
                },
              ],
              "yAxes": Array [
                Object {
                  "barPercentage": 0.2,
                  "maintainAspectRatio": false,
                  "scaleLabel": Object {
                    "display": true,
                    "labelString": "Votes",
                  },
                },
              ],
            },
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
